src/config.h

```diff
@@ -309,4 +309,14 @@ int pthread_setname_np(const char *name);
 void setcpuaffinity(const char *cpulist);
 #endif
 
+/* deprecate unsafe functions
+ *
+ * NOTE: We do not use the poison pragma since it
+ * will error on stdlib definitions in files as well*/
+#if (defined GCC_VERSION && GCC_VERSION >= 5) && !defined __APPLE__
```

Comment:
should be `__GNUC__`?

Reply:
> should be `__GNUC__`?

I don't think so since these attributes are compiler dependent.

Reply:
`__GNUC__` is also dependent on compiler.
My ubuntu can't recognize GCC_VERSION, not sure I'm wrong.
Maybe you can see the following code for GCC version.
https://github.com/redis/redis/blob/599e59ebc57283f52c60a8de56ec5f44d053109a/src/server.c#L5288